akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerSpec.scala

```diff
@@ -666,10 +666,11 @@ class CircuitBreakerSpec extends AkkaSpec with BeforeAndAfter with MockitoSugar
 
     "reset failure count after exception in call" when {
       "exception is defined as Success" in {
-        val breaker = CircuitBreakerSpec.multiFailureCb()
-        breaker().withCircuitBreaker(Future(sayHi))
-        for (n ← 1 to 4) breaker().withCircuitBreaker(Future(throwException))
-        awaitCond(breaker().currentFailureCount == 4, awaitTimeout)
+        val breaker: CircuitBreakerSpec.Breaker = CircuitBreakerSpec.multiFailureCb()
+        breaker().withCircuitBreaker(Future.successful(sayHi))
```

Comment:
isn't this just slightly less racy, but still racy? `f.onComplete` is used inside the CircuitBreaker

Is this first successful needed at all, or is the test still valid without it?

Reply:
afaik oncomplete for a completed future executes on the calling thread but you're right we don't need any successes before the failures. Will remove